config/translate.go

```diff
@@ -165,25 +180,53 @@ func TranslateFromV1(old v1.Config) types.Config {
 	return config
 }
 
+// golang--
+func translateV1MkfsOptionsToV2_1OptionSlice(opts v1.MkfsOptions) []types.Option {
+	newOpts := make([]types.Option, len(opts))
+	for i, o := range opts {
+		newOpts[i] = types.Option(o)
+	}
+	return newOpts
+}
+
+// golang--
+func translateStringSliceToV2_1SSHAuthorizedKeySlice(keys []string) []types.SSHAuthorizedKey {
+	newKeys := make([]types.SSHAuthorizedKey, len(keys))
+	for i, k := range keys {
+		newKeys[i] = types.SSHAuthorizedKey(k)
+	}
+	return newKeys
+}
+
+// golang--
+func translateStringSliceToV2_1UsercreateGroupSlice(groups []string) []types.UsercreateGroup {
+	var newGroups []types.UsercreateGroup
+	for _, g := range groups {
+		newGroups = append(newGroups, types.UsercreateGroup(g))
+	}
+	return newGroups
+}
+
 func TranslateFromV2_0(old v2_0.Config) types.Config {
 	translateVerification := func(old v2_0.Verification) types.Verification {
 		var ver types.Verification
 		if old.Hash != nil {
-			h := types.Hash(*old.Hash)
+			// .String() here is a wrapper around MarshalJSON, which will put the hash in quotes
+			h := strings.Trim(old.Hash.String(), "\"")
```

Comment:
A comment about the old and new format would make this logic a little clearer.

Switching to the `fmt.Sprintf` above would also make it a bit clearer for me, and hey it removes one instance of reflection too (via less json).

Your explanation makes sense though and at this point I'm just nitting.